runtime/src/qio/qio.c

```diff
@@ -1778,6 +1782,17 @@ qioerr _qio_channel_final_flush_unlocked(qio_channel_t* ch)
         if( max_written > ch->end_pos ) max_written = ch->end_pos;
 
         if( method == QIO_METHOD_MMAP ) {
+          // We're going to truncate the file.
+
+          // Destroy the buffer now so that we munmap
+          // before we truncate the file. That is required
+          // for Windows/Cygwin but probably a good idea elsewhere too.
+          if( qbuffer_is_initialized(&ch->buf) ) {
+            // Destroy the buffer.
+            destroy_buffer_error = qbuffer_destroy(&ch->buf);
+            destroyed_buffer = 1;
```

Comment:
no - destroy_buffer_error is just the error code we got when calling qbuffer_destroy. It's only set once.